test/test_bit_vector.cpp

```diff
@@ -1,190 +1,199 @@
 
-        for (size_t r = 0; r < ones.size(); ++r) {
-            for (size_t k = 0; k < std::min(size_t(256), size_t(ones.size() - r)); ++k) {
-                pisa::bit_vector::unary_enumerator ee(e);
-                ee.skip(k);
-                uint64_t pos = ee.next();
-                MY_REQUIRE_EQUAL(ones[r + k], pos,
-                                 "r = " << r << " k = " << k);
+            for (size_t r = 0; r < ones.size(); ++r) {
+                uint64_t pos = e.next();
+                MY_REQUIRE_EQUAL(ones[r], pos, "r = " << r);
             }
-            e.next();
         }
-    }
 
-    {
-        pisa::bit_vector::unary_enumerator e(bitmap, 0);
-
-        for (size_t r = 0; r < ones.size(); ++r) {
-            for (size_t k = 0; k < std::min(size_t(256), size_t(ones.size() - r)); ++k) {
-                pisa::bit_vector::unary_enumerator ee(e);
-                uint64_t pos_skip = ee.skip_no_move(k);
-                uint64_t pos = ee.next();
-                MY_REQUIRE_EQUAL(ones[r], pos,
-                                 "r = " << r << " k = " << k);
-                MY_REQUIRE_EQUAL(ones[r + k], pos_skip,
-                                 "r = " << r << " k = " << k);
+        {
+            pisa::bit_vector::unary_enumerator e(bitmap, 0);
 
+            for (size_t r = 0; r < ones.size(); ++r) {
+                for (size_t k = 0; k < std::min(size_t(256), size_t(ones.size() - r)); ++k) {
+                    pisa::bit_vector::unary_enumerator ee(e);
+                    ee.skip(k);
+                    uint64_t pos = ee.next();
+                    MY_REQUIRE_EQUAL(ones[r + k], pos, "r = " << r << " k = " << k);
+                }
+                e.next();
             }
-            e.next();
         }
-    }
-
-    {
-        pisa::bit_vector::unary_enumerator e(bitmap, 0);
 
-        for (size_t pos = 0; pos < v.size(); ++pos) {
-            uint64_t skip = 0;
-            for (size_t d = 0; d < std::min(size_t(256), size_t(v.size() - pos)); ++d) {
-                if (v[pos + d] == 0) {
-                    pisa::bit_vector::unary_enumerator ee(bitmap, pos);
-                    ee.skip0(skip);
+        {
+            pisa::bit_vector::unary_enumerator e(bitmap, 0);
 
-                    uint64_t expected_pos = pos + d;
-                    for (; !v[expected_pos] && expected_pos < v.size(); ++expected_pos);
-                    if (!v[expected_pos]) break;
+            for (size_t r = 0; r < ones.size(); ++r) {
+                for (size_t k = 0; k < std::min(size_t(256), size_t(ones.size() - r)); ++k) {
+                    pisa::bit_vector::unary_enumerator ee(e);
+                    uint64_t pos_skip = ee.skip_no_move(k);
                     uint64_t pos = ee.next();
-                    MY_REQUIRE_EQUAL(expected_pos, pos,
-                                     "pos = " << pos << " skip = " << skip);
+                    MY_REQUIRE_EQUAL(ones[r], pos, "r = " << r << " k = " << k);
```

Comment:
@amallia You know you can use `CAPTURE` macro to print a variable if the test fails, like so:
```cpp
CAPTURE(r);
CAPTURE(k);
```
Or did you choose this over it because of something I'm not seeing?